lib/rubocop/config.rb

```diff
@@ -385,8 +385,9 @@ def file_to_include?(file)
     def allowed_camel_case_file?(file)
       file_to_include?(file) do |pattern, relative_path, absolute_path|
         pattern.to_s =~ /[A-Z]/ &&
+          (relative_path.end_with?('.gemspec') ||
```

Comment:
I'd move this to the very beginning of the file and add some comment explaining why the check is needed. I'd probably use `File.extname` instead of `end_with?` - seem more intention-revealing to me.